src/Umbraco.Web.BackOffice/Security/PasswordChanger.cs

```diff
@@ -1,107 +1,104 @@
+        public PasswordChanger(ILogger<PasswordChanger<TUser>> logger)
         {
             _logger = logger;
         }
 
         /// <summary>
         /// Changes the password for a user based on the many different rules and config options
         /// </summary>
-        /// <param name="currentUser">The user performing the password save action</param>
-        /// <param name="savingUser">The user who's password is being changed</param>
-        /// <param name="passwordModel"></param>
-        /// <param name="userMgr"></param>
-        /// <returns></returns>
+        /// <param name="changingPasswordModel">The changing password model</param>
+        /// <param name="userMgr">The identity manager to use to update the password</param>
+        /// Create an adapter to pass through everything - adapting the member into a user for this functionality
+        /// <returns>The outcome of the password changed model</returns>
         public async Task<Attempt<PasswordChangedModel>> ChangePasswordWithIdentityAsync(
-            IUser currentUser,
-            IUser savingUser,
-            ChangingPasswordModel passwordModel,
-            IBackOfficeUserManager userMgr)
+            ChangingPasswordModel changingPasswordModel,
+            IUmbracoUserManager<TUser> userMgr)
         {
-            if (passwordModel == null) throw new ArgumentNullException(nameof(passwordModel));
-            if (userMgr == null) throw new ArgumentNullException(nameof(userMgr));
+            if (changingPasswordModel == null)
+            {
+                throw new ArgumentNullException(nameof(changingPasswordModel));
+            }
+
+            if (userMgr == null)
+            {
+                throw new ArgumentNullException(nameof(userMgr));
+            }
 
-            if (passwordModel.NewPassword.IsNullOrWhiteSpace())
+            if (changingPasswordModel.NewPassword.IsNullOrWhiteSpace())
             {
                 return Attempt.Fail(new PasswordChangedModel { ChangeError = new ValidationResult("Cannot set an empty password", new[] { "value" }) });
             }
 
-            var backOfficeIdentityUser = await userMgr.FindByIdAsync(savingUser.Id.ToString());
-            if (backOfficeIdentityUser == null)
+            var userId = changingPasswordModel.Id.ToString();
+            TUser identityUser = await userMgr.FindByIdAsync(userId);
+            if (identityUser == null)
             {
-                //this really shouldn't ever happen... but just in case
+                // this really shouldn't ever happen... but just in case
                 return Attempt.Fail(new PasswordChangedModel { ChangeError = new ValidationResult("Password could not be verified", new[] { "oldPassword" }) });
             }
 
-            //Are we just changing another user's password?
-            if (passwordModel.OldPassword.IsNullOrWhiteSpace())
+            // Are we just changing another user/member's password?
```

Comment:
@Shazwazza @bergmania I've just merged from dev as there was a conflict, this line been added but I am relying on this logic being done in the calling controller now:

   ```
         if (!currentUser.IsAdmin() && savingUser.IsAdmin())
            {
                return Attempt.Fail(new PasswordChangedModel { ChangeError = new ValidationResult("The current user cannot change the password for the specified user", new[] { "resetPassword" }) });
```

Reply:
I will update this logic in the calling controller

Reply:
@bergmania I have updated this logic in `UsersController`. This `PasswordChanger` is also used in 'CurrentUserController and 'MembersController'. I don't think the same admin logic applies there, but please let me know if it should.

Reply:
(I suggest someone who made the security fix tests this branch against the same criteria)

Reply:
Works as expected